client/client.go

```diff
@@ -172,6 +176,10 @@ func (c *Client) Run(ctx context.Context) error {
 	// https://jira.percona.com/browse/PMM-4245
 
 	oneDone := make(chan struct{}, 3)
```

Comment:
should'nt this be bumped to 4?
Also why we are not using wait groups?

Reply:
Yep, it should 👍 . Fixed.

Here is a trick, once one goroutine terminated this function returns immediately, and on upper-level context killed, and because of that other goroutines start terminating.